internal/sys/file_table.go

```diff
@@ -0,0 +1,141 @@
+package sys
+
+import "math/bits"
+
+// FileTable is a data structure mapping 32 bit file descriptor to file objects.
```

Comment:
move the heft of this to /RATIONALE.md then link back to it. e.g. See /RATIONALE.md. this way the godoc can focus on what it is doing overall vs how and why it was made the way it was.